dm/loader/util.go

```diff
@@ -112,44 +115,57 @@ func getDBAndTableFromFilename(filename string) (string, string, error) {
 	return fields[0], fields[1], nil
 }
 
-func getMydumpMetadata(cli *clientv3.Client, cfg *config.SubTaskConfig, workerName string) (string, string, error) {
-	metafile := filepath.Join(cfg.LoaderConfig.Dir, "metadata")
-	loc, _, err := dumpling.ParseMetaData(metafile, cfg.Flavor)
-	if err != nil {
-		if os.IsNotExist(err) {
-			worker, err2 := getLoadTask(cli, cfg.Name, cfg.SourceID)
-			if err2 != nil {
-				log.L().Warn("get load task", log.ShortError(err2))
-			}
-			if worker != "" && worker != workerName {
-				return "", "", terror.ErrLoadTaskWorkerNotMatch.Generate(worker, workerName)
-			}
-		}
-		if terror.ErrMetadataNoBinlogLoc.Equal(err) {
-			log.L().Warn("dumped metadata doesn't have binlog location, it's OK if DM doesn't enter incremental mode")
-			return "", "", nil
+func getMydumpMetadata(ctx context.Context, cli *clientv3.Client, cfg *config.SubTaskConfig, workerName string) (string, string, error) {
+	metafile := "metadata"
+	failpoint.Inject("TestRemoveMetaFile", func() {
+		err := storage.RemoveAll(ctx, cfg.LoaderConfig.Dir, nil)
+		if err != nil {
+			log.L().Warn("TestRemoveMetaFile Error", log.ShortError(err))
 		}
-
-		toPrint, err2 := os.ReadFile(metafile)
+	})
+	loc, _, err := dumpling.ParseMetaData(ctx, cfg.LoaderConfig.Dir, metafile, cfg.Flavor)
+	if err == nil {
+		return loc.Position.String(), loc.GTIDSetStr(), nil
+	}
+	if storage.IsNotExistError(err) {
+		failpoint.Inject("TestRemoveMetaFile", func() {
+			panic("success check file not exist!!")
+		})
+		worker, err2 := getLoadTask(cli, cfg.Name, cfg.SourceID)
 		if err2 != nil {
-			toPrint = []byte(err2.Error())
+			log.L().Warn("get load task", log.ShortError(err2))
+		}
+		if worker != "" && worker != workerName {
+			return "", "", terror.ErrLoadTaskWorkerNotMatch.Generate(worker, workerName)
 		}
```

Comment:
I mean we can always return here. `IsNotExistError` will not generate more meaningful message if continue

Reply:
ok, i see. i will return here.